version: "3"

services:
    db: 
        container_name: django-db
        image: mysql:5.7
        networks:
            - django-net
        volumes:
            - django-db:/var/lib/mysql
            - ./config/db/mysql/initdb.d:/docker-entrypoint-initdb.d
            - ./config/db/mysql/my.cnf:/etc/my.cnf
        ports: 
            - "3306:3306"
        environment:
            MYSQL_ROOT_PASSWORD: root
            MYSQL_DATABASE: django
            MYSQL_USER: django
            MYSQL_PASSWORD: django
            TZ: Asia/Tokyo
        restart: always

    app:
        container_name: django-app
        depends_on: 
            - db
        build:
            context: .
            dockerfile: config/docker/Dockerfile
        networks:
            - django-net
        ports:
            - "8000:80"
        volumes:
            - ./src:/var/www/html
            - ./logs:/var/www/logs
        tty: true
        environment:
            DJANGO_DB_HOST: db
            DJANGO_DB_NAME: django
            DJANGO_DB_USER: django
            DJANGO_DB_PASSWORD: django
        privileged: true

    phpmyadmin:
        container_name: django-phpmyadmin
        image: phpmyadmin/phpmyadmin
        networks:
            - django-net
        environment:
            PMA_ARBITRARY: 1
            PMA_HOSTS: db
            PMA_USER: root
            PMA_PASSWORD: root
        links:
            - db
        ports:
            - "8001:80"

networks: 
    django-net:
volumes: 
    django-db: